action.go

```diff
@@ -57,8 +67,9 @@ func (b *BaseAction) PreMachine(context *DebosContext,
 }
 func (b *BaseAction) PreNoMachine(context *DebosContext) error { return nil }
 func (b *BaseAction) Run(context *DebosContext) error          { return nil }
-func (b *BaseAction) Cleanup(context DebosContext) error       { return nil }
-func (b *BaseAction) PostMachine(context DebosContext) error   { return nil }
+func (b *BaseAction) Cleanup(context *DebosContext) error      { return nil }
```

Comment:
Ok lets see if i got this right; As i can't really work it out entirely from your commit message

* Cleanup() gets called only if the Run  for an action was started and in the same machine (host or fake) as Run has run
* PostCleanup() gets call for all actions regardless and always on the host
* Due to all the deferring both *Cleanup methods get called in reverse order; Cleanup always happens first (either on host or fake).

You can't use PostMachine as that doesn't get called on failure.

For some reason this doesn't sit right with me (mainly because the order is getting a bit complex ;)).. I think it would help if we name things so it's a bit more clear when it gets called? I found it quite hard to work out what hte differences between the two cleanup methods are.

Maybe make it `CleanupRun` and `PostMachineCleanup` ?